### PR TITLE
[CDAP-6177] XMLParser transform plugin

### PR DESCRIPTION
**Pull request by ananya-bhattacharya**
The XMLParser transform plugin receives structured record from XMLReader source plugin and creates structured record based on the xpath specified by the user for each field.
Wiki link for Requirements and Design : https://wiki.cask.co/display/CE/XML+Parser

All review comments from https://github.com/caskdata/hydrator-plugins/pull/293 are addressed in this PR.

### CODE REVIEW

**Comment by gokulavasan** [comment]
LGTM 👍

**Comment by prinam** [comment]
Hey @ananya-bhattacharya, as @gokulavasan mentioned earlier today, this PR didn't have a Travis build. Can you build on Travis manually and paste the results here? And next time, you can push a change to the branch to trigger the build?

File: transform-plugins/src/main/java/co/cask/hydrator/plugin/XMLParser.java [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+          builder.set(fieldName, null);
+        } else {
+          builder.set(fieldName, TypeConvertor.get(value, type));
+        }
+      }
+      emitter.emit(builder.build());
+    } catch (Exception e) {
+      switch (config.processOnError) {
+        case EXIT_ON_ERROR:
+          throw new IllegalStateException("Terminating process on error: " + e.getMessage());
```
Comment (anwar6953) [inline transform-plugins/src/main/java/co/cask/hydrator/plugin/XMLParser.java]: pass `e` as the second parameter, so the entire stack trace is not lost.

File: transform-plugins/docs/XMLParser-transform.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    +=========================================================================================================+
+    | category  | title              | year   |  price  | subcategory                                         |
+    +=========================================================================================================+
+    | cooking   | Everyday Italian   | null   |  null   | <subcategory><type>Continental</type><genre>European|
+    |           |                    |        |         | cuisines</genre></subcategory>                      |
+    | children  | Harry Potter       | 2005   | 49.99   | <subcategory><type>Series</type><genre>fantasy      |
+    |           |                    |        |         | literature</genre></subcategory>                    |
+    +=========================================================================================================+
+
+Here, since the subcategory contains child nodes, the plugin will retrun the complete subcategory node (along with its
```
Comment (anwar6953) [inline transform-plugins/docs/XMLParser-transform.md]: "retrun" --> "rerun" ?

File: transform-plugins/docs/XMLParser-transform.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+child elements) as string as ``<subcategory><type>Continental</type><genre>European cuisines</genre></subcategory>`` .
+This is to ensure that the plugin returns a sigle XML event for a structured record intead of the two child events:
```
Comment (anwar6953) [inline transform-plugins/docs/XMLParser-transform.md]: "sigle" --> "single"?

File: transform-plugins/docs/XMLParser-transform.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+**processOnError:** The action to take in case of an error.
+                     - "Ignore error and continue"
+                     - "Exit on error" : Stops processing upon encountering an error
+                     - "Write to error dataset" :  Writes the error record to an error dataset and continues
+
+Example
+-------
+
+This example parses an XML record received in the "body" field of the input record following the *xPathMappings* for
+each field name. The output structured record will be created  using the type specified for each field in the
```
Comment (anwar6953) [inline transform-plugins/docs/XMLParser-transform.md]: nit: extra space after "created"

**Comment by anwar6953** [comment]
Just had a few minor (easy to address) comments.

**Comment by anwar6953** [comment]
lgtm.
Can you post a link to the travis build, please?

**Comment by ananya-bhattacharya** [comment]
Travis build link : https://travis-ci.org/caskdata/hydrator-plugins/builds/144376696

**Comment by sreevatsanraman** [comment]
Build passed. Merging.

**Pull request merged**